### DISCUSSION TITLE
Reusable providers?

### DISCUSSION THREAD

**Discussion by b4ltz** [rrousselGit/riverpod#1209] [Q&A]
Hi is it possible to reuse a provider? 

For example:
```dart 
final queryProvider = StateProvider((ref)=>'');
```

then override it in different Routes/Page:
```dart
MaterialPageRoute(
      builder: (_) => ProviderScope(
        overrides: [
          queryProviderDisposable.overrideWithValue(StateController('')),
        ],
        child: const Page1(),
      ),
);
```

**Comment by rrousselGit** [comment]
Yes, your code works.

**Reply by b4ltz** [reply]
Thanks! on another note, when do i need to specify the ```dependecies``` property of a provider? because when i tried to add a ```queryProvider``` to ```providerA```, all providers that depend on ```providerA``` gets an error like this:

```
The provider providerXXX#4477b, but it specified a 'dependendencies' list yet that list does not contain providerxxxxx
To fix, add providerxxxxx to providerXXXX
````

Do I really need to add all the dependencies that a provider depends on when its root provider has other dependency?
